WordPress/gutenberg
# Revert fix/slug-to-classname

This reverts https://github.com/WordPress/gutenberg/pull/32352 that by trying to fix https://github.com/WordPress/gutenberg/issues/32347 created issues https://github.com/WordPress/gutenberg/issues/32719 and https://github.com/WordPress/gutenberg/issues/32721

Third parties (such as Jetpack) rely on specific behaviours from lodash's `kebabCase` method (see the table at https://github.com/WordPress/gutenberg/issues/32347). Because the generated classes are attached to the markup we can't really remove kebabCase as that's the source of the behaviour in use. It seems the way to fix this is by implementing the same behavior around numbers and numerals in the server.

While we do that we can gain some time by reverting https://github.com/WordPress/gutenberg/pull/32352 

## How to test

― Test 1

- Install and activate Jetpack on a site. This requires a public site. You can created one at jurassic.ninja for example.
- Download the gutenberg plugin from this PR (see artifacts in the CI tab). 
- Activate the subscription form block with [these instructions](https://jetpack.com/support/jetpack-blocks/subscription-form-block/).
- Insert the following code into code view of editor.

```html

<div class="wp-block-jetpack-subscriptions wp-block-jetpack-subscriptions__supports-newline">[jetpack_subscription_form show_subscribers_total="false" button_on_newline="false" custom_background_button_color="#151516" custom_font_size="16" custom_border_radius="3" custom_border_weight="1" custom_border_color="#ffffff" custom_padding="15" custom_spacing="10" submit_button_classes="has-16-font-size has-ffffff-border-color" email_field_classes="has-16-font-size has-ffffff-border-color" show_only_email_and_button="true"]</div>

```
- Update post.
- The block should display as valid.

― Test 2

- Using TT1-blocks update the `green` slug to `green2green` in the `theme.json`.
- Go to the editor and attach that color text to a block. Verify that the generated class is `has-green-2-green-color`.
- Go to the front-end and inspect the contents of the `global-styles-inline-css` style tag. Verify that
  - the generated classes are `has-green2green-color`, `has-green2green-background-color` and `has-green2green-border-color` 
  - the CSS custom property is `--wp--preset--color--green2green`.
- Load the site editor and verify that it behaves like the front end (search for `has-green2green*` classes, etc.
